examples/trials/benchmarking/automlbenchmark/nni/extensions/NNI/architectures/run_random_forest.py

```diff
@@ -110,33 +120,35 @@ def run_random_forest(dataset, config, tuner, log):
     intermediate_best_scores = []           # should be monotonically increasing 
     
     while True:
-        try:
-            trial_count += 1
+        try:            
             param_idx, cur_params = tuner.generate_parameters()
-            train_params = cur_params.copy()
-            if 'TRIAL_BUDGET' in cur_params:
-                train_params.pop('TRIAL_BUDGET')
-            if cur_params['max_leaf_nodes'] == 0: 
-                train_params.pop('max_leaf_nodes')
-            if cur_params['max_depth'] == 0:
-                train_params.pop('max_depth')
-            log.info("Trial {}: \n{}\n".format(param_idx, cur_params))
+            if cur_params is not None and cur_params != {}:
```

Comment:
do we need to wait for `else`?

Reply:
Currently, if `else` is true, the loop just continue to the next iteration, and the `trial_count` is not incremented. The loop exits when the trial limit is reached or the tuner throws an exception.